apps/framework-cli/src/infrastructure/redis/new_redis_client.rs

```diff
@@ -0,0 +1,1049 @@
+        if let Some((lock_key, _)) = locks.get(name) {
+            let result = self
+                .leadership_manager
+                .has_lock(
+                    self.connection_manager.connection.clone(),
+                    lock_key,
+                    &self.instance_id,
+                )
+                .await?;
+            return Ok(result);
+        }
+        Ok(false)
+    }
+
+    // --------------------------------------
+    // Periodic Tasks and Connection Monitor
+    // -------------------------------
+    pub fn start_periodic_tasks(&self) {
+        // Spawn a periodic presence update task
+        let connection = self.connection_manager.connection.clone();
+        let key_prefix = self.config.key_prefix.clone();
+        let instance_id = self.instance_id.clone();
+        let presence_task = tokio::spawn(async move {
+            let mut interval = tokio::time::interval(tokio::time::Duration::from_secs(1));
+            loop {
+                interval.tick().await;
+                let key = format!("{}::{}::presence", key_prefix, instance_id);
+                let now = std::time::SystemTime::now()
+                    .duration_since(std::time::UNIX_EPOCH)
+                    .map(|d| d.as_secs())
+                    .unwrap_or(0);
+                let mut conn_guard = connection.write().await;
+                let _: redis::RedisResult<()> = redis::cmd("SETEX")
+                    .arg(&key)
+                    .arg(3)
+                    .arg(now)
+                    .query_async(&mut *conn_guard)
+                    .await;
+            }
+        });
+
+        // Save the presence task handle
+        let mut presence_guard = self.presence_task.lock().unwrap();
+        *presence_guard = Some(presence_task);
+        drop(presence_guard);
+
+        let connection_manager = self.connection_manager.clone();
+        let config = self.config.clone();
+        if let Ok(mut guard) = self.connection_monitor_task.lock() {
+            *guard = Some(tokio::spawn(async move {
+                let mut interval = tokio::time::interval(tokio::time::Duration::from_secs(5));
+                loop {
+                    interval.tick().await;
+                    if !connection_manager.ping().await {
+                        log::error!(
+                            "<NewRedisClient> Redis connection lost, attempting reconnection"
+                        );
+                        let client = redis::Client::open(config.url.clone())
+                            .expect("Failed to open client during reconnection");
+                        connection_manager
+                            .attempt_reconnection(&client, &config)
+                            .await;
+                    }
+                }
+            }));
+        } else {
+            log::error!("<NewRedisClient> Failed to lock connection_monitor_task");
+        }
+
+        let create_monitor_task = || {
```

Comment:
Can you explain the difference between the task above and this one? Feels like they are both doing pings to reconnect